glib/src/object.rs

```diff
@@ -52,8 +52,6 @@ pub unsafe trait ObjectType:
     type GlibType: 'static;
     /// type of the FFI Class structure.
     type GlibClassType: 'static;
-    /// type of the Rust Class structure.
-    type RustClassType: 'static;
```

Comment:
So far we didn't require there to be a wrapper type for `ObjectSubclass` impls... maybe that's something we can also change in the future as it seems like it would make things easier, or not?

Also it would allow returning the type itself instead of the parent type from `ObjectSubclass::get_instance()` and similar weirdnesses that we have right now.

What do you think @ids1024? We can do that separate from this PR though, no need to block on this here.

Reply:
That could be useful. Similarly to `get_instance`, all of the `*Impl` traits have methods receiving the parent type, which a lot of implementations will immediately want to downcast.

I've been thinking of that recently. I wonder if subclassing should requiring defining a `glib_wrapper!` and specifying that as an associated type in `ObjectSubclass`. If that doesn't lead to recursive type errors, and doesn't make basic subclassing too hard, it could improve some of these things.

But yeah, that's separate and doesn't need to block this PR.

Reply:
Yeah passing it to the virtual methods was also what I was thinking of, that should be possible then. Can you create an issue for this?

Reply:
Opened https://github.com/gtk-rs/gtk-rs/issues/14

Reply:
Thanks!